indy-besu/docs/design/did-doc.md

```diff
@@ -0,0 +1,261 @@
+
+* id (string): A string with a format did:[indy2|indy|sov]:<namespace>:<id>#<key-alias> or #<key-alias>
+* controller: A string with fully qualified DID. DID must exist.
+* type (string)
+* publicKeyJwk (string, optional): A map representing a JSON Web Key that conforms to RFC7517. See definition of publicKeyJwk for additional constraints.
+  > **NOTE, that `publicKeyJwk` string representation does not match to the specification and should be converted into object on the VDR level.**
+* publicKeyMultibase (string, optional): A base58-encoded string that conforms to a MULTIBASE encoded public key.
+
+Note: A single verification method entry cannot contain both publicKeyJwk and publicKeyMultibase, but must contain at least one of them.
+
+#### Verification Relationship
+
+Verification relationship expresses the relationship between the DID subject and a verification method. This property is optional.
+
+* id (optional): reference to verification method's id.
+* verificationMethod (optional): contains full information about verification method.
+
+#### Services
+
+Services can be defined in a DIDDoc to express means of communicating with the DID subject or associated entities.
+
+* id (string): The value of the id property for a Service MUST be a URI conforming to RFC3986. A conforming producer MUST NOT produce multiple service entries with the same ID. A conforming consumer MUST produce an error if it detects multiple service entries with the same ID. It has a follow formats: did:[indy2|indy|sov]:<namespace>:<id>#<service-alias> or #<service-alias>.
+* type (string): The service type and its associated properties SHOULD be registered in the DID Specification Registries
+* serviceEndpoint (strings): A string that conforms to the rules of RFC3986 for URIs, a map, or a set composed of a one or more strings that conform to the rules of RFC3986 for URIs and/or maps.
+* accept ([string], optional): An array of media types in the order of preference for sending a message to the endpoint. These identify a profile of DIDComm Messaging that the endpoint supports.
+* routingKeys ([string], optional): An ordered array of strings referencing keys to be used when preparing the message for transmission as specified in Sender Process to Enable Forwarding, above.
+
+#### DID Document metadata
+
+Each DID Document MUST have a metadata section when a representation is produced. It can have the following properties:
+
+* created (timestamp): Time of a block ordered a transaction for DID Doc creation
+* updated (timestamp): The updated field is null if an Update operation has never been performed on the DID document. Time of a block ordered a transaction changed a DID Doc last time
+* deactivated (string): If DID has been deactivated, DID document metadata MUST include this property with the boolean value true. By default this is set to false.
+
+## Transactions (Smart Contract's methods)
+
+Contract name: **DidRegistry** 
+
+### Create DID
+
+* Method: `createDid`
```

Comment:
Trust guarantees are achieved by generating DID identifier from the initial ed25519 public key of the DID
An appropriate verification key needs signature verification, but ed25519 signature is not as trivial for Solidity as EcDSA. Due to this, we have postponed its implementation to the next phases.

Reply:
If I do a fast forward to the next phase here - is there currently a rough idea how to get that verification working? I find this quite important, as self-certification was good property of old `did:sov`/`did:indy` and `indy2` is aiming to port/recreate them.

Thinking out loud, I suppose the entity registering DID would also have to additionally supply:
- full public key
- signature of some data (what data though? perhaps the passed diddoc structure?)

Is that right?

One more question is also whether this next phase is meant to be done is prior or after deployment of `did:indy2`. 
- If it's done prior - cool, the DID IDs could maintain the nice property they hade before. 
- If it's done after - well, the data DIDs migrated from older ledger would preserve the property of being "self-certifying", but not the new DIDs (assuming the on-chain verification is not yet in place and deferred as future work). So that result in kind of "mixed bag", some `did:indy2` DIDs being self-certifying, but some not.

Reply:
There are 2 ways to support ed25519: check it in the smart contract or enable this signature along with EcDSA in Ethereum.
It makes sense to sign the entire transaction.
And the key will be stored in the Verification method of the Document DID, as in Indy.
The community offers various options for implementing ed25519 and is actively discussing its inclusion in standard Ethereum signatures. 
https://ethresear.ch/t/verify-ed25519-signatures-cheaply-on-eth-using-zk-snarks/13139

Therefore, we just postponed this task a little in the hope that the community will come to a final decision.

Regarding implementation time, this task is scheduled for phase 2 ([roadmap](https://github.com/hyperledger/indy-node/blob/3ca041604284c6232a606c9084dbc2cf3c3e1632/indy-besu/docs/roadmap.md#phase-2-beta)). After phase 2 is implemented, we will be able to recommend using this code for production deployment.